insights/parsers/package_provides_java.py

```diff
@@ -2,48 +2,52 @@
-Through this way, when there is jdk bug, we can detect whether a running java application will be affected.
+    This parser reads the output from pre_command::
 
+        ``for jp in `/bin/ps auxwww | grep java | grep -v grep| awk '{print $11}' | sort -u`; do echo $jp `readlink -e $jp | xargs rpm -qf`; done``
 
-Typical contents of the pre_command::
+    and coverts it into a dict that links running_java_command with its package name.
+    The reason why using above pre_command is that we need to record the links between running_java_command
+    and package which provides the java command. In ``ps aux`` output, we can only get what java command starts a
+    java application, instead of java package.
+    Through this way, when there is jdk bug, we can detect whether a running java application will be affected.
 
-    /usr/lib/jvm/jre/bin/java java-1.8.0-openjdk-headless-1.8.0.141-3.b16.el6_9.x86_64
 
-Parsed result::
+    Typical contents of the pre_command::
 
-    {"command": "/usr/lib/jvm/jre/bin/java", "package": "java-1.8.0-openjdk-headless-1.8.0.141-3.b16.el6_9.x86_64"}
+        /usr/lib/jvm/jre/bin/java java-1.8.0-openjdk-headless-1.8.0.141-3.b16.el6_9.x86_64
 
-Examples:
+    Parsed result::
 
-    >>> command_package = shared[PackageProvidesJava]
-    >>> command_package.data
-    {"command": "/usr/lib/jvm/jre/bin/java", "package": "java-1.8.0-openjdk-headless-1.8.0.141-3.b16.el6_9.x86_64"}
-    >>> command_package.get("command")
-    '/usr/lib/jvm/jre/bin/java'
+        self.command = "/usr/lib/jvm/jre/bin/java"
+        self.package = "java-1.8.0-openjdk-headless-1.8.0.141-3.b16.el6_9.x86_64"
+
+    Examples:
+
+        >>> command_package = shared[PackageProvidesJava]
+        >>> command_package.command
+        '/usr/lib/jvm/jre/bin/java'
+        >>> command_package.package
+        'java-1.8.0-openjdk-headless-1.8.0.141-3.b16.el6_9.x86_64'
 """
 
-from insights import parser, Parser, LegacyItemAccess
+from insights import parser, Parser
 
 
 @parser('package_provides_java')
-class PackageProvidesJava(Parser, LegacyItemAccess):
+class PackageProvidesJava(Parser):
     """
-    Parse the output of pre_command
-    ``for jp in `/bin/ps auxwww | grep java | grep -v grep| awk '{print $11}' | sort -u`; \
-    do echo $jp `readlink -e $jp | xargs rpm -qf`; done``.
+    Parse the output of pre_command::
+
+        ``for jp in `/bin/ps auxwww | grep java | grep -v grep| awk '{print $11}' | sort -u`; do echo "$jp `readlink -e $jp | xargs rpm -qf`"; done``.
+
+    Attributes:
+        command (str): The java command that starts application.
+        package (str): Java package that provides above java command.
     """
 
     def parse_content(self, content):
-        comamnd_package_dict = {}
-        for line in content:
-            l = line.split()
-            if len(l) == 2:
-                comamnd_package_dict = ({"command": l[0], "package": l[1]})
-        self.data = comamnd_package_dict
+        l = content[0].split()
```

Comment:
I'm sorry for the late comment, you should process the situation when there is not Java is running on some hosts. In this case, the pre_command will return nothing.

Better raise a ParseException for it. for the docstring, pls refer to`insights/parsers/proc_limits.py`